src/main/java/wolox/training/models/Book.java

```diff
@@ -57,6 +63,7 @@ public String getGenre() {
     }
 
     public void setGenre(String genre) {
+        Preconditions.checkArgument(genre != null && !genre.isEmpty(), "Genre must have a valid value");
```

Comment:
You can replace this check with

```
import org.apache.commons.lang.StringUtils;
 StringUtils.isEmpty ()
```
It is a method that performs both checks.

```suggestion
        Preconditions.checkArgument(!StringUtils.isEmpty(genre), "Genre must have a valid value");
```
You could take all the sets that have a string as a parameter, delete the checkNotNull and place this check.

Reply:
Update preconditions to setters with string argument